narwhals/expr.py

```diff
@@ -873,7 +963,8 @@ def map_batches(
         return self.__class__(
             lambda plx: self._to_compliant_expr(plx).map_batches(
                 function=function, return_dtype=return_dtype
-            )
+            ),
+            is_order_dependent=True,  # safest assumption
```

Comment:
what if we add an extra param to the function for the user to determine 🤔?

Reply:
i think this function is only useful/implemented for eager backends anyway, so for now not sure we should bother